File: vcpkg-dependency-source/carbon-blueexposure-c855175f5f261a301a12b63c0337e8eb8d92b081/tests/python/BlueExposureTest/test/test_optionalargs.py
# Copyright (c) 2026 CCP Games

import unittest
import BlueExposureTest


class TestOptionalArgs(unittest.TestCase):
    """
    A set of test cases focusing on optional arguments to methods.
    """

    def testMethodCallsAndReturnValues(self):
        """
        This tests the methods with all supported data types. Each method takes
        in one parameter and simply returns it. This exercises the exposure code
        for each data type, both as a parameter and a return value.
        """

        x = BlueExposureTest.TestOptionalArgs()

        orgVal = False
        val = x.MethodReturningBool(orgVal)
        self.assertEqual(val, orgVal)
        val = x.MethodReturningBool()
        self.assertEqual(val, False)

        orgVal = 42
        val = x.MethodReturningInt(orgVal)
        self.assertEqual(val, orgVal)
        val = x.MethodReturningInt()
        self.assertEqual(val, 0)

        orgVal = 42
        val = x.MethodReturningUnsignedInt(orgVal)
        self.assertEqual(val, orgVal)
        val = x.MethodReturningUnsignedInt()
        self.assertEqual(val, 0)

        orgVal = 3.14
        val = x.MethodReturningFloat(orgVal)
        self.assertAlmostEqual(val, orgVal, places=6)
        val = x.MethodReturningFloat()
        self.assertAlmostEqual(val, 0.0, places=6)

        orgVal = 2.71828182845904523536
        val = x.MethodReturningDouble(orgVal)
        self.assertEqual(val, orgVal)
        val = x.MethodReturningDouble()
        self.assertEqual(val, 0.0)

        orgVal = "this is a test"
        val = x.MethodReturningString(orgVal)
        self.assertEqual(val, orgVal)
        val = x.MethodReturningString()
        self.assertEqual(val, "")

        orgVal = "this is a test"
        val = x.MethodReturningConstChar(orgVal)
        self.assertEqual(val, orgVal)
        val = x.MethodReturningConstChar()
        self.assertEqual(val, "")

        orgVal = u"this is a test"
        val = x.MethodReturningWString(orgVal)
        self.assertEqual(val, orgVal)
        val = x.MethodReturningWString()
        self.assertEqual(val, u"")

        orgVal = u"this is a test"
        val = x.MethodReturningConstWChar(orgVal)
        self.assertEqual(val, orgVal)
        val = x.MethodReturningConstWChar()
        self.assertEqual(val, u"")

        orgVal = (1, 2)
        val = x.MethodReturningVector2(orgVal)
        self.assertEqual(val, orgVal)
        val = x.MethodReturningVector2()
        self.assertEqual(val, (0, 0))

        orgVal = (1, 2, 3)
        val = x.MethodReturningVector3(orgVal)
        self.assertEqual(val, orgVal)
        val = x.MethodReturningVector3()
        self.assertEqual(val, (0, 0, 0))

        orgVal = object()
        val = x.MethodReturningPyObject(orgVal)
        self.assertEqual(val, orgVal)
        val = x.MethodReturningPyObject()
        self.assertEqual(val, None)

        orgVal = BlueExposureTest.TestMultipleInterfaces()
        val = x.MethodReturningBlueObject(orgVal)
        self.assertEqual(val, orgVal)
        val = x.MethodReturningBlueObject()
        self.assertEqual(val, None)

    def testOptionalArgs(self):
        x = BlueExposureTest.TestOptionalArgs()

        self.assertRaises(TypeError, x.CalcSum)
        self.assertRaises(TypeError, x.CalcSum, None)

        result = x.CalcSum(1)
        self.assertEqual(result, 1)

        result = x.CalcSum(1, 2)
        self.assertEqual(result, 3)

        result = x.CalcSum(1, 2, 3)
        self.assertEqual(result, 6)

        result = x.CalcSum(1, 2, 3, 4)
        self.assertEqual(result, 10)

        result = x.CalcSum(1, 2, 3, 4, 5)
        self.assertEqual(result, 15)

        result = x.CalcSum(1, 2, 3, 4, 5, 6)
        self.assertEqual(result, 21)

        result = x.CalcSum(1, 2, 3, 4, 5, 6, 7)
        self.assertEqual(result, 28)

        result = x.CalcSum(1, 2, 3, 4, 5, 6, 7, 8)
        self.assertEqual(result, 36)

        result = x.CalcSum(1, 2, 3, 4, 5, 6, 7, 8, 9)
        self.assertEqual(result, 45)

        self.assertRaises(TypeError, x.CalcSum, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10)

    def testBeOptionalArgs(self):
        x = BlueExposureTest.TestOptionalArgs()

        result = x.BeOptionalIsSpecified(1)
        self.assertEqual(result, True)

        result = x.BeOptionalIsSpecified()
        self.assertEqual(result, False)

        result = x.BeOptional()
        self.assertEqual(result, 0)

        result = x.BeOptional(0)
        self.assertEqual(result, 0)

        result = x.BeOptional(1)
        self.assertEqual(result, 1)

        result = x.BeOptionalWithDefault()
        self.assertEqual(result, 123)

        result = x.BeOptionalWithDefault(0)
        self.assertEqual(result, 0)

        result = x.BeOptionalWithDefault(1)
        self.assertEqual(result, 1)

File: vcpkg-dependency-sources-29731196920/tests/TestOptionalArgs.cpp
////////////////////////////////////////////////////////////////////////////////
//
// Creator:		Snorri Sturluson
// Created:		February 2013
// Copyright (c) 2026 CCP Games
//

#include "TestOptionalArgs.h"

bool TestOptionalArgs::MethodReturningBool( bool val )
{
	return val;
}

int TestOptionalArgs::MethodReturningInt( int val )
{
	return val;
}

uint32_t TestOptionalArgs::MethodReturningUnsignedInt( uint32_t val )
{
	return val;
}

float TestOptionalArgs::MethodReturningFloat( float val )
{
	return val;
}

double TestOptionalArgs::MethodReturningDouble( double val )
{
	return val;
}

std::string TestOptionalArgs::MethodReturningString( const std::string& val )
{
	return val;
}

const char* TestOptionalArgs::MethodReturningConstChar( const char* val )
{
	return val;
}

std::wstring TestOptionalArgs::MethodReturningWString( const std::wstring& val )
{
	return val;
}

const wchar_t* TestOptionalArgs::MethodReturningConstWChar( const wchar_t* val )
{
	return val;
}

Vector2 TestOptionalArgs::MethodReturningVector2( const Vector2& val )
{
	return val;
}

Vector2d TestOptionalArgs::MethodReturningVector2d( const Vector2d& val )
{
	return val;
}

Vector3 TestOptionalArgs::MethodReturningVector3( const Vector3& val )
{
	return val;
}

Vector3d TestOptionalArgs::MethodReturningVector3d( const Vector3d& val )
{
	return val;
}

Vector3i TestOptionalArgs::MethodReturningVector3i( const Vector3i& val )
{
	return val;
}

Vector4d TestOptionalArgs::MethodReturningVector4d( const Vector4d& val )
{
	return val;
}

#if BLUE_WITH_PYTHON
PyObject* TestOptionalArgs::MethodReturningPyObject( PyObject* val )
{
	return val;
}
#endif


TestMultipleInterfaces* TestOptionalArgs::MethodReturningBlueObject( TestMultipleInterfaces* val )
{
	return val;
}

int TestOptionalArgs::CalcSum( int a1, int a2, int a3, int a4, int a5, int a6, int a7, int a8, int a9 )
{
	return a1 + a2 + a3 + a4 + a5 + a6 + a7 + a8 + a9;
}

bool TestOptionalArgs::BeOptionalIsSpecified( Be::Optional<int> val )
{
	return val.IsAssigned();
}

int TestOptionalArgs::BeOptional( Be::Optional<int> val )
{
	if( val.IsAssigned() )
	{
		return val;
	}
	else
	{
		return 0;
	}
}

int TestOptionalArgs::BeOptionalWithDefault( Be::OptionalWithDefaultValue<int, 123> val )
{
	return val;
}
